File: 2015/day05/day05.py
from re import search, findall

with open('day05.input') as file:
    print(sum(bool(2 < len(findall('[aeiou]', line))
                   and search(r'(\w)\1', line)
                   and not search('ab|cd|pq|xy', line)) for line in file))
    file.seek(0)
    print(sum(bool(search(r'(..).*\1', line) and
                   search(r'(.).\1', line)) for line in file))

File: 2015/day05/day05.nim
import re

var
  i = 0
  j = 0

for line in lines "day05.input":
  if findAll(line, re"[aeiou]").len > 2 and
     contains(line, re"(\w)\1") and
     not contains(line, re"ab|cd|pq|xy"):
    inc i
  if contains(line, re"(..).*\1") and
     contains(line, re"(.).\1"):
    inc j

echo i
echo j
